modAionImpl/src/org/aion/zero/impl/valid/TXValidator.java

```diff
@@ -47,12 +47,13 @@
             Collections.synchronizedMap(new LRUMap<>(128 * 1024));
 
     public static boolean isValid(AionTransaction tx) {
-        Boolean valid = cache.get(ByteArrayWrapper.wrap(tx.getHash()));
+        ByteArrayWrapper transactionHash = (tx.getTransactionHash() == null) ? null : ByteArrayWrapper.wrap(tx.getTransactionHash());
+        Boolean valid = cache.get(transactionHash);
```

Comment:
if transactionHash == null, this line might throw NPE

Reply:
I believe the solution here should be that if the transactionHash is null, to compute `isValid0` on the transaction and return the result without added anything to the cache, otherwise, Kelvin is right, we could end up with a null key pointing to an answer that other null transactionHash transactions will inherit.

But this change is consensus-breaking, and we probably want to open a separate issue for this and do it at the same time as another fork.

For now, I will simply take care of the NPE.

Reply:
I don't believe a null transactionHash is possible. The check probably isn't necessary at all. The `getTransactionHash()` method cannot return null. Agreed?